File: ComiAPI/ComiCore/Model/Movies.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace ComiCore.Model
{
    public class Movies : BaseEntity
    {
        public string Name { get; set; }
        public string Description { get; set; }
        public DateTime Manufacture { get; set; }
        public string Status { get; set; }
        public int Episode { get; set; }
        public string Path { get; set; }
        public Quality Quality { get; set; }
        public DateTime Time { get; set; }
        public string Country { get; set; }
        public Language Language { get; set; }
        public Kind Kind { get; set; }
        public virtual ICollection<GenreMovies> GenresMovies { get; set; }
        public virtual ICollection<MoviesDirector> MoviesDirectors { get; set; }
        public virtual ICollection<MoviesActor> MoviesActors { get; set; }
        public virtual ICollection<Trailer> Trailers { get; set; }
        public virtual ICollection<Comment> Comments { get; set; }
    }
    public enum Quality
    {
        HD,
        HDRip
    }
    public enum Language
    {
        [Display(Name = "Tiếng Việt")]
        Vietnamese,
        [Display(Name = "Thuyết minh")]
        ThuyetMinh,
        [Display(Name = "Lồng tiếng")]
        LongTieng,
        VietSub
    }
    public enum Kind
    {
        [Display(Name = "Phim lẻ")]
        PhimLe,
        [Display(Name = "Phim bộ")]
        PhimBo,
        [Display(Name = "Phim chiếu rạp")]
        PhimChieuRap
    }
}

File: ComiMovies/src/app/models/movies.ts
import { BaseEntity } from './ientity';
import { Director } from './director';
import { Actor } from './actor';

export class Movies extends BaseEntity {
    title: string
    name: string
    description: string
    releaseDate: Date
    status: string
    episode: number
    path: string
    quality: string
    time: Date
    country: string
    language: Language
    kind: Number
    genreMovies: Array<any>
    moviesDirectors: Array<any>
    moviesActors: Array<any>
    trailers: Array<any>
    comments: Array<any>
}

export const QUALITY  = [
    {code: "HD", displayName:  "HD"},
    {code: "HDRip", displayName:  "HDRip"},
    {code: "HQ", displayName:  "HQ"},
    {code: "TWO_K", displayName:  "2K"},
    {code: "FOUR_K", displayName:  "4K"},
]

export enum Language {
    ORIGINAL = "Nguyên bản",
    THUYETMINH = "Thuyết Minh",
    LONGTIENG = "Lồng Tiếng",
    VIETSUB = "Viet subtitle"
}

export const Kind = [
    { code: "PhimLe", name: "Phim lẽ" },
    { code: "PhimBo", name: "Phim bộ" },
    { code: "PhimChieuRap", name: "Phim chiếu rạp" },
]

export const DIRECTORS:Director[] = [
    new Director(1, "Steven Spielberg", ""),
    new Director(2, "Alfred Hitchcock", ""),
    new Director(3, "Martin Scorsese", ""),
    new Director(4, "Christopher Nolan", ""),
    new Director(5, "James Cameron", ""),
    new Director(6, "Francis Ford Coppola", ""),
    new Director(7, "Quentin Tarantino", ""),
    new Director(8, "Guy Ritchie", ""),
]

export const ACTORS:Actor[] = [
    new Actor(1, "Jason Statham", ""),
    new Actor(2, "Sylvester Stallone", ""),
    new Actor(3, "Martin Scorsese", ""),
    new Actor(4, "Christopher Nolan", ""),
    new Actor(5, "James Cameron", ""),
    new Actor(6, "Francis Ford Coppola", ""),
    new Actor(7, "Quentin Tarantino", ""),
    new Actor(8, "Guy Ritchie", ""),
]